File: javatest/omnia/data/cache/MemoizedLongTest.java
package omnia.data.cache;

import static com.google.common.truth.Truth.assertThat;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class MemoizedLongTest {

  @Test
  public void memoize_didReturnSimpleLongMemoizer() {
    assertThat(MemoizedLong.memoize(() -> 132L)).isInstanceOf(SimpleLongMemoizer.class);
  }
}

File: tests/omnia/data/cache/MemoizedLongTest.kt
package omnia.data.cache

import com.google.common.truth.Truth.assertThat
import kotlin.test.Test

class MemoizedLongTest {

  @Test
  fun memoize_didReturnSimpleLongMemoizer() {
    assertThat(MemoizedLong.memoize { 132L }).isInstanceOf(SimpleLongMemoizer::class.java)
  }
}